app/dashboard/utils.py

```diff
@@ -0,0 +1,194 @@
+    standardbounties_addr = getStandardBountiesContractAddresss(network)
+    bounty_abi = json.loads(standardbounties_abi)
+    getBountyContract = web3.eth.contract(standardbounties_addr, abi=bounty_abi)
+    return getBountyContract
+
+
+def get_bounty(bounty_enum, network):
+    try:
+        ipfs = getIPFS()
+        standard_bounties = getBountyContract(network)
+        issuer, deadline, fulfillmentAmount, paysTokens, bountyStage, balance = standard_bounties.functions.getBounty(bounty_enum).call()
+        bountydata = standard_bounties.functions.getBountyData(bounty_enum).call()
+        arbiter = standard_bounties.functions.getBountyArbiter(bounty_enum).call()
+        token = standard_bounties.functions.getBountyToken(bounty_enum).call()
+        fulfillments = []
+        fulfill_enum = 0
+        more_fulfillments = True
+        while more_fulfillments:
+            try:
+                accepted, fulfiller, data = standard_bounties.functions.getFulfillment(bounty_enum, fulfill_enum).call()
+                fulfill_enum += 1
+                fulfillments.append({
+                    'accepted': accepted,
+                    'fulfiller': fulfiller,
+                    'data': json.loads(ipfs.cat(data)),
+                    })
+            except BadFunctionCallOutput:
+                more_fulfillments = False
+
+        bounty = {
+            'id': bounty_enum,
+            'issuer': issuer,
+            'deadline': deadline,
+            'fulfillmentAmount': fulfillmentAmount,
+            'paysTokens': paysTokens,
+            'bountyStage': bountyStage,
+            'balance': balance,
+            'data': json.loads(ipfs.cat(bountydata)),
+            'arbiter': arbiter,
+            'token': token,
+            'fulfillments': fulfillments,
+            'network': network,
+        }
+        return bounty
+    except BadFunctionCallOutput:
+        raise BountyNotFoundException
+
+
+# processes a bounty returned by get_bounty
+def process_bounty(bounty_data):
+    did_change, old_bounty, new_bounty = process_bounty_details(bounty_data)
+
+    if did_change:
+        print("- processing changes")
+        process_bounty_changes(old_bounty, new_bounty, None)
+
+    return did_change, old_bounty, new_bounty
+
+
+def has_tx_mined(txid, network):
+    web3 = getWeb3(network)
+    try:
+        transaction = web3.eth.getTransaction(txid)
+        return transaction.blockHash != HexBytes('0x0000000000000000000000000000000000000000000000000000000000000000')
+    except:
+        return False
+
+
+def getBountyID(issueURL, network):
```

Comment:
Looks like this tries the db first, and then web3.  What happens if they get out of sync or something?  Should it try to reconcile the db and web3 before returning anything?

Reply:
> Looks like this tries the db first, and then web3.

yeah, for performance reasons this is _way_ faster

>  What happens if they get out of sync or something?

Given that the bounty ids increment on the smart contract... I dont see how they could get out of sync.   

Maybe if there was a new version of standardbounties.. but in that case we'd have to migrate the old bounties over to that contract and we'd incur a migration cost anyway.